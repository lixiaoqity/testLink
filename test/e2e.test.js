/* eslint-disable no-undef */
const execa = require('execa');
const { describe, test, expect } = require('@jest/globals');
expect.addSnapshotSerializer(require('jest-snapshot-serializer-ansi'));

describe('end-to-end integration', () => {
	test('prints tool version with arument -v', async () => {
		const { stdout, stderr } = await execa('node', ['index', '-v']);
		expect(stdout).toMatchSnapshot();
		expect(stderr).toEqual('');
	});

	test('prints url information with a path', async () => {
		const { stdout, stderr } = await execa('node', ['index', 'test4.txt']);
		expect(stdout).toMatchSnapshot();
		expect(stderr).toEqual('');
	});

	test('prints url information with two path', async () => {
		const { stdout, stderr } = await execa('node', [
			'index',
			'test4.txt',
			'test5.html',
		]);
		expect(stdout).toMatchSnapshot();
		expect(stderr).toEqual('');
	});

	test('prints url information with path and json output', async () => {
		const { stdout, stderr } = await execa('node', [
			'index',
			'-j',
			'test5.html',
		]);
		expect(stdout).toMatchSnapshot();
		expect(stderr).toEqual('');
	});

	test('prints url information with ignore arugment', async () => {
		const { stdout, stderr } = await execa('node', [
			'index',
			'-i',
			'test5.html',
			'test4.txt',
		]);
		expect(stdout).toMatchSnapshot();
		expect(stderr).toEqual('');
	});
});
