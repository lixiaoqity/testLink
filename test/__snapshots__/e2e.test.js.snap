// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`end-to-end integration prints tool version with arument -v 1`] = `
"name: testLinks
version: 1.0.0"
`;

exports[`end-to-end integration prints url information with a path 1`] = `"https://github.com is good."`;

exports[`end-to-end integration prints url information with ignore arugment 1`] = `
"This is invalid.  It doesn't use http:// or https://
<html>
</html>
"
`;

exports[`end-to-end integration prints url information with path and json output 1`] = `"{\\"url\\":\\"https://github.com\\",\\"status\\":200}"`;

exports[`end-to-end integration prints url information with two path 1`] = `
"https://github.com is good.
https://github.com is good."
`;
